akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala

```diff
@@ -89,10 +89,19 @@ private[http] object DeflateCompressor {
 class DeflateDecompressor(maxBytesPerChunk: Int = Decoder.MaxBytesPerChunkDefault) extends DeflateDecompressorBase(maxBytesPerChunk) {
 
   override def createLogic(attr: Attributes) = new DecompressorParsingLogic {
-    override val inflater: Inflater = new Inflater()
+    var maybeNoWrappedInflater: Option[Inflater] = None
+    override def inflater: Inflater = maybeNoWrappedInflater.getOrElse(new Inflater())
```

Comment:
Hmm, that construction is a bit weird. Maybe just like that:

```scala
var _inflater: Inflater = _
def inflater = if (_inflater ne null) _inflater else throw new IllegalStateException("Inflater used before probing for wrapping")
```

Reply:
Yeah wasn't sure if you're were ok with nulls - will change it round :)